Src/Sarif.PatternMatcher/SearchSkimmer.cs

```diff
@@ -991,8 +990,7 @@ private void RunMatchExpressionForSingleLineRegexes(AnalyzeContext context, Matc
 
                 if (_validators != null && matchExpression.IsValidatorEnabled)
                 {
-                    string ruleName = matchExpression.Name ?? reportingDescriptor.Name;
-                    IEnumerable<ValidationResult> validationResults = _validators.Validate(ruleName,
+                    IEnumerable<ValidationResult> validationResults = _validators.Validate(reportingDescriptor.Id,
```

Comment:
>reportingDescriptor.Id [](http://example.com/codeflow?start=91&length=22)

why use `reportingDescriptor.Id` here, not `matchExpression.Id` like other 2 changes?